### ISSUE TITLE
[NativeAot] Test failure in System.Reflection.Context

### ISSUE THREAD

**Issue by VSadov** [dotnet/runtime#83361]
This happens when running libraries tests with NativeAot. It seems recent.

```
ILC : error IL3003: System.Reflection.Context.Delegation.DelegatingAssembly.Location.get: Member 'System.Reflection.Context.Delegation.DelegatingAssembly.Location.get' with 'RequiresAssemb
lyFilesAttribute' overrides base member 'System.Reflection.Assembly.Location.get' without 'RequiresAssemblyFilesAttribute'. 'RequiresAssemblyFilesAttribute' annotations must match across a
ll interface implementations or overrides. [C:\dotnet001\runtime\src\libraries\System.Reflection.Context\tests\System.Reflection.Context.Tests.csproj]
```

**Comment by MichalStrehovsky** [comment]
It's from @vitek-karas change to generate this warning. There is a fix for this in #83250 and also in #83253 but code reviews have slowed down to a crawl. (I inlined the fix in my PR because I needed to see the extra-platforms results and don't want to wait a day until someone reviews a standalone PR that fixes this. I really wish I could merge test changes without a review.)

**Comment by MichalStrehovsky** [comment]
Fixed in the above pull request.
